src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableSortedSet_1.Enumerator.cs

```diff
@@ -139,13 +139,13 @@ object System.Collections.IEnumerator.Current
             /// </summary>
             public void Dispose()
             {
-                _root = null;
+                _root = null!;
                 _current = null;
-                Stack<RefAsValueType<Node>> stack;
+                Stack<RefAsValueType<Node>>? stack;
                 if (_stack != null && _stack.TryUse(ref this, out stack))
                 {
                     stack.ClearFastWhenEmpty();
-                    s_enumeratingStacks.TryAdd(this, _stack);
+                    s_enumeratingStacks.TryAdd(this, _stack!);
```

Comment:
Not sure why compiler evaluating _stack can be null, might be a bug

Reply:
Presumably because of that `ref this ` above; in theory the TryUse call could zero out `this` such that `_stack` would be `null`.